src/nvim/mbyte.c

```diff
@@ -994,37 +994,37 @@ int utf_char2len(const int c)
 int utf_char2bytes(const int c, char *const buf)
 {
   if (c < 0x80) {  // 7 bits
-    buf[0] = c;
+    buf[0] = (char)c;
     return 1;
   } else if (c < 0x800) {  // 11 bits
-    buf[0] = 0xc0 + ((unsigned)c >> 6);
-    buf[1] = 0x80 + (c & 0x3f);
+    buf[0] = (char)(0xc0 + ((unsigned)c >> 6));
+    buf[1] = (char)(0x80 + (unsigned)(c & 0x3f));
```

Comment:
I think it's better to convert `c` to `unsigned` before the bitwise operation line done in the line above this one.